vcd/resource_vcd_nsxt_firewall.go

```diff
@@ -0,0 +1,328 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceVcdNsxtFirewall() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdNsxtFirewallCreateUpdate,
+		ReadContext:   resourceVcdNsxtFirewallRead,
+		UpdateContext: resourceVcdNsxtFirewallCreateUpdate, // Update is exactly the same operation as create
+		DeleteContext: resourceVcdNsxtFirewallDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdNsxtFirewallImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"edge_gateway_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge Gateway ID in which Firewall rule is located",
```

Comment:
"Firewall rule" is a bit confusing here (and in the previous comment). As if we're talking about a single rule, but in fact we're talking about the whole firewall object. Maybe change "rule" -> "rules"? Or some other suggestions?

Reply:
Yes, indeed. This was probably typo/leftover from initial version. Data Source had it right (plural). Fixed.